src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -176,20 +175,27 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            DateTime now = DateTime.Now;
-            await InvokeJobFunction(now, false);
-
             // restart the timer with the next schedule occurrence
-            now = DateTime.Now;
-            DateTime nextOccurrence = _schedule.GetNextOccurrence(now);
-            TimeSpan nextInterval = nextOccurrence - now;
+            await InvokeJobFunction(DateTime.Now, false);
+            TimeSpan nextInterval = ScheduleStatus.Next - DateTime.Now;
```

Comment:
That actually didn't change -- if you look at the previous iteration, we would update now with a new DateTime.Now after invoking. It confused me to cache that value but not actually re-use it so I switched it to use DateTime.Now directly.

In order for this interval to be correct, we need to calculate it after the function has run -- as close to the timer starting as possible. Example -- if you have an hour timer invoked at 12:00 and the invocation takes 1 minute, we want to calculate the interval for the next timer using 12:01 rather than at 12:00. Otherwise, you'd start a 1 hour timer at 12:01. But we really want it to be a 59-minute timer.

---
In reply to: [73020609](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/133#discussion_r73020609) [](ancestors = 73020609)